File: binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: TreeNode) -> list[int]:
        return self.dfs(root, [])
    
    def dfs(self, root: TreeNode, node_list: list[int]) -> list[int]:
        if not root:
            return []
        self.dfs(root.left, node_list)
        node_list.append(root.val)
        self.dfs(root.right, node_list)
        return node_list

File: binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution 
{
    public List<Integer> inorderTraversal(TreeNode root) 
    {
        List<Integer> nodeList = new ArrayList<Integer>();
        if (root == null)
            return nodeList;
        return DFS(root, nodeList);
    }
    
    public List<Integer> DFS(TreeNode root, List<Integer> nodeList)
    {
        if (root == null)
            return nodeList;
        DFS(root.left, nodeList);
        nodeList.add(root.val);
        DFS(root.right, nodeList);
        return nodeList;
    }
}